lib/vector/Vlib/copy.c

```diff
@@ -641,10 +653,12 @@ int Vect_copy_tables(struct Map_info *In, struct Map_info *Out, int field)
                 _("Unable to copy table <%s> for layer %d from <%s> to <%s>"),
                 Fi->table, Fi->number, Vect_get_full_name(In),
                 Vect_get_name(Out));
+            Vect_destroy_map_struct(Out);
+            Vect_destroy_field_info(Fi);
```

Comment:
Here CID 1270330. Result of Vect_get_full_name() (L642) need to be freed.
No need to destroy `Out` or `Fi` here.